Epsilon.Abstractions/Component/KpiMatrixCollection.cs

```diff
@@ -0,0 +1,148 @@
+            headerRow.AppendChild(new TableRowProperties(new TableRowHeight
+            {
+                Val = (UInt32Value)(uint)headerRowHeight,
+            }));
+
+            // Empty top-left cell.
+            headerRow.AppendChild(CreateTableCellWithBorders("2500", new Paragraph(new Run(new Text("")))));
+
+            foreach (var assignment in module.KpiMatrix.Assignments)
+            {
+                var cell = CreateTableCellWithBorders("100");
+                cell.FirstChild.Append(new TextDirection
+                {
+                    Val = TextDirectionValues.BottomToTopLeftToRight,
+                });
+
+                cell.Append(new Paragraph(new Run(new Text(assignment.Name))));
+                headerRow.AppendChild(cell);
+            }
+
+            table.AppendChild(headerRow);
+
+            // Add the outcome rows.
+            foreach (var outcome in module.KpiMatrix.Outcomes.OrderBy(static o => o.Id))
+            {
+                var row = new TableRow();
+
+                // Add the outcome title cell.
+                row.AppendChild(CreateTableCellWithBorders("2500", new Paragraph(new Run(new Text(outcome.Title)))));
+
+                // Add the assignment cells.
+                foreach (var assignment in module.KpiMatrix.Assignments)
+                {
+                    var outcomeAssignment = assignment.Outcomes.FirstOrDefault(o => o.Title == outcome.Title);
+                    var cell = CreateTableCellWithBorders("100");
+
+                    // Set cell color based on GradeStatus.
+                    if (outcomeAssignment != null)
+                    {
+                        var fillColor = outcomeAssignment.GradeStatus switch
+                        {
+                            GradeStatus.Approved => "44F656",
+                            GradeStatus.Insufficient => "FA1818",
+                            GradeStatus.NotGraded => "FAFF00",
+                            _ => "FFFFFF",
+                        };
+                        cell.FirstChild.Append(new Shading
+                        {
+                            Fill = fillColor,
+                        });
+                    }
+
+                    // Add an empty text element since we're using color instead of text.
+                    cell.Append(new Paragraph(new Run(new Text(""))));
+                    row.AppendChild(cell);
+                }
+
+                table.AppendChild(row);
+            }
+
+            body.AppendChild(table);
+            body.AppendChild(new Paragraph(new Run(new Break
+            {
+                Type = BreakValues.Page,
+            })));
+        }
+
+        return body;
+    }
+
+    private static TableCell CreateTableCellWithBorders(string? width, params OpenXmlElement[] elements)
+    {
+        var cell = new TableCell();
+        var cellProperties = new TableCellProperties();
+        var borders = new TableCellBorders(
```

Comment:
Create alternating colours for each row and column so that it can be more easily read.